pkg/processor/sourceprocessor/source_category_filler_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2021 Sumo Logic, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sourceprocessor
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+func TestNewSourceCategoryFiller(t *testing.T) {
+	cfg := createDefaultConfig().(*Config)
+	cfg.SourceCategory = "qwerty-%{k8s.namespace.name}-%{k8s.pod.uid}"
+
+	filler := newSourceCategoryFiller("_sourceCategory", cfg)
+
+	assert.Equal(t, filler.attributeName, "_sourceCategory")
+	assert.Equal(t, 2, len(filler.templateAttributes))
+	assert.Equal(t, "k8s.namespace.name", filler.templateAttributes[0])
+	assert.Equal(t, "k8s.pod.uid", filler.templateAttributes[1])
+}
+
+func TestFill(t *testing.T) {
+	cfg := createDefaultConfig().(*Config)
+	cfg.SourceCategory = "source-%{k8s.namespace.name}-%{k8s.pod.uid}-cat"
+
+	attrs := pdata.NewAttributeMap()
```

Comment:
A suggestion: we can also use `pdata.NewAttributeMapFromMap(attrMap map[string]pdata.AttributeValue)` in order to initialize from map

Reply:
Yeah I considered that, but I fail to see how this code:

```
	attrs := pdata.NewAttributeMapFromMap(map[string]pdata.AttributeValue{
		"k8s.namespace.name": pdata.NewAttributeValueString("ns-1"),
		"k8s.pod.uid":        pdata.NewAttributeValueString("123asd"),
	})
```

is clearer than:

```
	attrs := pdata.NewAttributeMap()
	attrs.InsertString("k8s.namespace.name", "ns-1")
	attrs.InsertString("k8s.pod.uid", "123asd")
```

I recognize that the `pdata.NewAttributeMapFromMap` function declares the whole map in one command, instead of in three commands, but the amount of boilerplate code is a bit much to me.